script.js

```diff
@@ -1,4 +1,227 @@
+  var deck = [];
+  var suits = ["hearts", "diamonds", "clubs", "spades"];
+  var suitIndex = 0;
+
+  var SPADE = "♠️";
+  var HEART = "♥️";
+  var CLUB = "♣️";
+  var DIAMOND = "♦️";
+
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+
+    var counter = 1;
+    while (counter <= 13) {
+      var rankCounter = counter;
+      var cardName = rankCounter;
+
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        rankCounter = 10;
+        cardName = "jack";
+      } else if (cardName == 12) {
+        rankCounter = 10;
+        cardName = "queen";
+      } else if (cardName == 13) {
+        rankCounter = 10;
+        cardName = "king";
+      }
+      if (currentSuit == "spades") {
+        currentEmoji = SPADE;
+      } else if (currentSuit == "hearts") {
+        currentEmoji = HEART;
+      } else if (currentSuit == "clubs") {
+        currentEmoji = CLUB;
+      } else if (currentSuit == "diamonds") {
+        currentEmoji = DIAMOND;
+      }
+
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      deck.push(card);
+
+      counter = counter + 1;
+    }
+    suitIndex = suitIndex + 1;
+  }
+
+  return deck;
+};
+
+// Get a random card from the deck
+var getRandomIndex = function (size) {
+  return Math.floor(Math.random() * size);
+};
+
+var shuffleCards = function (cards) {
+  var index = 0;
+
+  while (index < cards.length) {
+    var randomIndex = getRandomIndex(cards.length);
+
+    var currentItem = cards[index];
+
+    var randomItem = cards[randomIndex];
+
+    cards[index] = randomItem;
+    cards[randomIndex] = currentItem;
+
+    index = index + 1;
+  }
+
+  return cards;
+};
+
+//Start
+var currentGameMode = "draw card";
+var deck = shuffleCards(makeDeck());
+var playerHand = [];
+var computerHand = [];
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (currentGameMode == "draw card") {
+    var counter = 0;
+    while (counter < 2) {
+      counter += 1;
+      var playerCardInHand = deck.pop();
+      playerHand.push(playerCardInHand);
+      var computerCardInHand = deck.pop();
+      computerHand.push(computerCardInHand);
+    }
+
+    var playerScore = scoreCounter(playerHand);
+    console.log(playerScore);
+    var computerScore = scoreCounter(computerHand);
+    console.log(computerScore);
+    var showPlayerHand = showHand(playerHand);
+    console.log(showPlayerHand);
+    var showComputerHand = showHand(computerHand);
+    console.log(showComputerHand);
+
+    var myOutputValue = "";
+    myOutputValue =
+      "Player drew: " +
+      showPlayerHand +
+      "<br>" +
+      "Player score is: " +
+      playerScore +
+      "<br>" +
+      "Computer drew " +
+      showComputerHand +
+      "<br>" +
+      "Computer score is:" +
+      computerScore;
+
+    currentGameMode = "hit or stand";
+    return myOutputValue;
```

Comment:
The next cheapest way to make your code more concise is to move your logic in the different game modes away from the `main` function, and into their own game mode functions. [Akira's video on refactoring code based on program states for game modes](https://basics.rocketacademy.co/5-managing-state-and-input-validation/8.2-program-state-for-game-modes#refactoring-code) explains it well, and you can review it if you are still unsure about this.

Let us practise doing this just for the "draw card" game mode. We know that in the "draw card" game mode we are going to have a function there, and the game mode's purpose is to print something into the grey output box. In this case, it is printing the current player hand and computer hand, as well as both of their scores.

Let's start by simplifying this like so:
```
...
if (currentGameMode == "draw card") {
  return drawCard();
}
...
```
This already cleans up the `main` function by cutting the number of lines it has, because we are moving the "draw card" logic into another function

Looking at above, we know that `drawCard()` replaces what we previously have as the `return` value, which is `myOutputValue` in [line 122](https://github.com/rocketacademy/basics-blackjack/blob/8357b1a693cd773c45088fe872a40486129fd6f1/script.js#L122). Since we have a "draw card" function now, we know that we want to replicate its previous behaviour, which is to change some output and have the `drawCard` function return a string to `main`, where `main` in turn returns it into the grey output box.

Let's set up the `drawCard` function by copying and pasting the previous content inside the `if (currentGameMode == "draw card") { ... }` block:
```
var drawCard = function () {
  var counter = 0;
  while (counter < 2) {
    counter += 1;
    var playerCardInHand = deck.pop();
    playerHand.push(playerCardInHand);
    var computerCardInHand = deck.pop();
    computerHand.push(computerCardInHand);
  }

  var playerScore = scoreCounter(playerHand);
  console.log(playerScore);
  var computerScore = scoreCounter(computerHand);
  console.log(computerScore);
  var showPlayerHand = showHand(playerHand);
  console.log(showPlayerHand);
  var showComputerHand = showHand(computerHand);
  console.log(showComputerHand);

  var myOutputValue = "";
  myOutputValue =
    "Player drew: " +
    showPlayerHand +
    "<br>" +
    "Player score is: " +
    playerScore +
    "<br>" +
    "Computer drew " +
    showComputerHand +
    "<br>" +
    "Computer score is:" +
    computerScore;

  currentGameMode = "hit or stand";
  return myOutputValue;
};
```
Let's take a look at which variables were previously created in `main`: these variables should then be passed in as parameters into the `drawCard` function. For variables that are created globally (outside of any function), it is not necessary to pass them in as function parameters.

As it turns out, you did not have any variable here that is meant only for `main`. Even for `myOutputValue`, if you move it into the new `drawCard` function, it will be scoped to the `drawCard` function when creating it there.

The only global variables used in `drawCard` are `currentGameMode` and `deck`, and since they are global variables, they need not be a part of the function parameters.